src/kudu/consensus/quorum_util.h

```diff
@@ -141,6 +141,19 @@ bool ShouldEvictReplica(const RaftConfigPB& config,
                         MajorityHealthPolicy policy,
                         std::string* uuid_to_evict = nullptr);
 
+// Helper function to compute the regional count from the config. If the
+// leader_uuid is present, it also figures out the region of the leader.
+void GetRegionalCountsFromConfig(
+    const RaftConfigPB& config, const std::string& leader_uuid,
+    std::map<std::string, int>* regional_count,
+    std::string* leader_region = nullptr);
+
+// Make each regions voter count, the max of voter distribution and current
+// voters
+void AdjustVoterDistributionWithCurrentVoters(
+    const RaftConfigPB& config,
+    std::map<std::string, int> *voter_distribution);
```

Comment:
I don't fully understand the proposal of updating Active Config. Active config is a set of peers. It already has
the current peers.
There is no change in flexi-raft protocol for this. It modifies voter_distribution which actually should have been
named target_voters/minimum_voters to begin with. FlexiRaft's majority calculation has not changed.

Reply:
I am suggesting to not change this within FlexiRaft code at all. When config changes are done --  addition / removal of members from the replicaset, why not modify the map there? we are modifying active config there already.
https://github.com/anirbanr-fb/kuduraft/blob/1.8.raft/src/kudu/consensus/metadata.proto#L239

If we do that, majority of this diff is not required.

Reply:
I don't think we should modify consensus metadata to keep it up to date with config. Firstly because it is not the same as voter list. When we remove voters from config we would not change it to go below 3. So it will only be done on addition and not on all removals. So it is already highly heursitical.

It will become a nightmare to debug why we are out of synch with what was provided to us by automation. 
If you want to add another field called target_voters_distribution_ and put it in metadata.proto then it would make sense, but then we would have to modify FlexiRaft to move to the new field.